File: frontend/src/types/subscription.ts
import type { Plan } from "./plan"

export interface Subscription {
  id: string
  status: 'active' | 'expired' | 'canceled' | 'pending'
  plan: Plan
  start_at: string
  end_at: string
  auto_renew: boolean
}

export interface CreditWallet {
  id: string
  total_credits: number
  used_credits: number
  remaining_credits: number
  period_start: string
  period_end: string
}

export interface UserSubscriptionResponse {
  subscription: Subscription | null
  credit_wallet: CreditWallet | null
  purchased_wallet: CreditWallet | null
}

export interface CreditActivity {
  id: string
  created_at: string
  activity_type: string
  amount: number
  tokens_used: number | null
  model_used: string | null
  llm_calls: number | null
  reason: string
  agent_name: string | null
  project_name: string | null
  story_title: string | null
  task_type: string | null
}

export interface CreditActivityResponse {
  total: number
  items: CreditActivity[]
  summary: {
    total_credits_spent: number
    total_tokens_used: number
    total_llm_calls: number
    top_agent: string | null
    top_model: string | null
  }
}

File: backend/app/schemas/subscription.py
"""Subscription and Credit Wallet Schemas"""

from datetime import datetime
from uuid import UUID
from pydantic import BaseModel

from app.schemas.plan import PlanPublic


class SubscriptionPublic(BaseModel):
    """Public subscription data"""
    id: UUID
    status: str  # 'active', 'expired', 'canceled', 'pending'
    start_at: datetime | None = None
    end_at: datetime | None = None
    auto_renew: bool = True
    plan: PlanPublic


class CreditWalletPublic(BaseModel):
    """Public credit wallet data"""
    id: UUID
    total_credits: int
    used_credits: int
    remaining_credits: int
    period_start: datetime | None = None
    period_end: datetime | None = None


class UserSubscriptionResponse(BaseModel):
    """User subscription with credit wallet"""
    subscription: SubscriptionPublic | None = None
    credit_wallet: CreditWalletPublic | None = None


class UpdateAutoRenew(BaseModel):
    """Update auto-renew setting"""
    auto_renew: bool
